test/function.localstack.test.ts

```diff
@@ -603,7 +548,129 @@ localstackTestSuite("functionStack", (testResource, _stack, _app) => {
     "started!"
   );
 
-  testFunctionResource(
+  test(
+    "tokens",
+    (parent) => {
+      const token = Token.asString("hello");
+      const obj = { iam: "object" };
+      const token2 = Token.asAny(obj);
+      const obj2 = { iam: Token.asString("token") };
+      const nestedToken = Token.asAny(obj2);
+      const numberToken = Token.asNumber(1);
+      const listToken = Token.asList(["1", "2"]);
+      const nestedListToken = Token.asList([Token.asString("hello")]);
+      return new Function(
+        parent,
+        "function",
+        localstackClientConfig,
+        async () => {
+          return {
+            string: token,
+            object: token2 as unknown as typeof obj,
+            nested: nestedToken as unknown as typeof obj2,
+            number: numberToken,
+            list: listToken,
+            nestedList: nestedListToken,
+          };
+        }
+      );
+    },
+    {
+      string: "hello",
+      object: { iam: "object" },
+      nested: { iam: "token" },
+      number: 1,
+      list: ["1", "2"],
+      nestedList: ["hello"],
+    }
+  );
+
+  test(
+    "serialize entire table",
+    (parent) => {
+      const table = new aws_dynamodb.Table(parent, "table", {
+        partitionKey: {
+          name: "key",
+          type: aws_dynamodb.AttributeType.STRING,
+        },
+      });
+      const get = $AWS.DynamoDB.GetItem;
+      table.addGlobalSecondaryIndex({
+        indexName: "testIndex",
+        partitionKey: {
+          name: "key",
+          type: aws_dynamodb.AttributeType.STRING,
+        },
+      });
+
+      const flTable = new Table(table);
+
+      return new Function(
+        parent,
+        "function",
+        localstackClientConfig,
+        async () => {
+          get({
+            TableName: flTable,
```

Comment:
Oh dang, this forces the entire table to be serialized?

Reply:
yes, its pretty easy to do that and with the refactor to not re-write the closures, we'll almost always serialize the entire constructs.

Reply:
Do we get a chance to rewrite it in the serializer?

Reply:
Yeah, we remove common Construct/Resource and CfnResource fields. Heuristic driving right now. #239

Reply:
Which of course means that `table.resource.globalSecondaryIndices.length` would fail in a `Function` at runtime.

Reply:
> Which of course means that `table.resource.globalSecondaryIndices.length` would fail in a `Function` at runtime.

I get that we erase the `globalSecondaryIndices` data but do we get a chance to inline the `.length` value?